trunner/tools/phoenix.py

```diff
@@ -22,6 +22,23 @@ def wait_for_dev(port, timeout=0):
             raise TimeoutError
 
 
+def wait_for_vid_pid(port_hint: str, timeout=0):
+    asleep = 0
+    port = None
+
+    while port is None:
+        time.sleep(0.01)
+        asleep += 0.01
+
+        for p in list_ports.grep(port_hint):
+            port = p
+
+        if timeout and asleep >= timeout:
+            raise PhoenixdError(f"Couldn't find USB device with vid/pid: '{port_hint}'. Timeout.") from TimeoutError
```

Comment:
this is a generic function (which might be reused not only in `Phoenixd` class so it should throw generic exception `TimeoutError`, not phoenixd-specific one